core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedDocServicePlugin.java

```diff
@@ -170,6 +169,13 @@ private static void addMethodInfo(Map<Class<?>, Set<MethodInfo>> methodInfos,
                 });
     }
 
+    private static TypeSignature getReturnTypeSignature(Method method) {
+        if (isKFunction(method) && !isReturnTypeNothing(method)) {
+            return toTypeSignature(kFunctionGenericReturnType(method));
+        }
```

Comment:
nit; should we call `kFunctionReturnType` if `Nothing` type for more accurate representation in our doc services?

i.e.

```suggestion
        if (isKFunction(method)) {
            if (isReturnTypeNothing(method)) {
                return toTypeSignature(kFunctionReturnType(method));
            }
            return toTypeSignature(kFunctionGenericReturnType(method));
        }
```

Reply:
Seems better to use `kFunctionReturnType` for `Nothing` as you commented. 🤔 
Let me update too.

Reply:
Seems better to use `kFunctionReturnType` for `Nothing`.
Let me update it too.

Reply:
Wouldn't this throw an exception, though? (And this was the motivation of this PR).

Reply:
But, the return type will be just `Void` not `Nothing` even if we do this.